src/controller_examples/rabbitmq_controller/exec/reconciler.rs

```diff
@@ -615,27 +619,38 @@ requires
     rabbitmq@.metadata.namespace.is_Some(),
 ensures
     config_map@ == rabbitmq_spec::update_server_config_map(rabbitmq@, found_config_map@),
-{
-    let mut owner_references = Vec::new();
-    owner_references.push(rabbitmq.controller_owner_ref());
-    proof {
-        assert_seqs_equal!(
-            owner_references@.map_values(|owner_ref: OwnerReference| owner_ref@),
-            rabbitmq_spec::make_role(rabbitmq@).metadata.owner_references.get_Some_0()
-        );
-    }
-    let mut metadata = found_config_map.metadata();
+{    
+    let mut config_map = found_config_map.clone();
+    let made_server_cm = make_server_config_map(rabbitmq);
 
-    // Since we requirement the owner_reference only contains current cr, this set operation won't change anything.
-    // Similarly, we never set finalizers for any stateful set, resetting finalizers won't change anything.
-    // The reason why we add these two operations is that it makes the proof easier.
-    // In this way, we can easily show that what the owner references and finalizers of the object in every update request
-    // for stateful set are.
-    metadata.set_owner_references(owner_references);
-    metadata.unset_finalizers();
-    found_config_map.set_data(make_server_config_map(rabbitmq).data().unwrap());
-    found_config_map.set_metadata(metadata);
-    found_config_map
+    config_map.set_data({
+        let old_data_opt = found_config_map.data();
+        let mut old_data = if old_data_opt.is_some() { old_data_opt.unwrap() } else { StringMap::empty() };
+        old_data.extend(made_server_cm.data().unwrap());
+        old_data
+    });
+    config_map.set_metadata({
+        let mut metadata = found_config_map.metadata();
+        // Since we requirement the owner_reference only contains current cr, this set operation won't change anything.
```

Comment:
I thought we only check owner reference for statefulset and return if the owner reference does not match. Do we do the same for the configmap?

Reply:
Yes. The invariants we need for config map are the same as stateful set.

Reply:
Can you remind me why we want to stop reconcile and return error when the found configmap has unmatching owner reference? I know we did so for statefulset to prevent that tricky bug.

Reply:
I don't think we have this. Pleas make sure you get it right.

Reply:
The comment here was copied from stateful set, so there should be some mistakes.

Reply:
But you answered yes to my first question, which was asking whether the controller checks owner reference for configmap.

Reply:
Yeah I realized that. Because this code block is for making config map, so I got the question wrong.

Reply:
So I guess the comment is wrong (since the configmap here might have an old owner reference)?

Reply:
The comment is wrong, but we did decide to set the owner references for config map.

Reply:
I think I can come up with a way to avoid setting the owner_references for updates.

Reply:
I remember that this decision is to make the proof easier (when proving invariants). I don't have a strong opinion on this so you can decide what is the best way.